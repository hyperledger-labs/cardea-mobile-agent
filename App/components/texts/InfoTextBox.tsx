import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { TextBoxTheme, TextTheme } from '../../theme'

export interface TextBoxProps {
  children: string
}

const iconSize = 30
const offset = 10

const style = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: TextBoxTheme.background,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: TextBoxTheme.border,
    padding: 10,
  },
  headerText: {
    ...TextTheme.normal,
    flexShrink: 1,
  },
  icon: {
    marginRight: offset,
  },
})

const InfoTextBox: React.FC<TextBoxProps> = ({ children }) => {
  return (
    <View style={[style.container]}>
      <View style={[style.icon]}>
        <Icon name={'info'} size={iconSize} color={TextBoxTheme.text} />
      </View>
      <Text
        style={[
          style.headerText,
          {
            fontWeight: 'bold',
          },
        ]}
      >
        {children}
      </Text>
    </View>
  )
}

export default InfoTextBox
