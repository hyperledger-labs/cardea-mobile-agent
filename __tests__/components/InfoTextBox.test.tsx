import React from 'react'
import { create } from 'react-test-renderer'

import InfoTextBox from '../../App/components/texts/InfoTextBox'

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')

describe('InfoTextBox Component', () => {
  it('Renders correctly', () => {
    const tree = create(<InfoTextBox>Hello World</InfoTextBox>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
