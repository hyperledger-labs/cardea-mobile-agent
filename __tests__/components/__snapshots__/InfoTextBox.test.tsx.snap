// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTextBox Component Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(53, 130, 63, 0.7)",
        "borderColor": undefined,
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "padding": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginRight": 10,
        },
      ]
    }
  >
    <Icon
      color="#FFFFFF"
      name="info"
      size={30}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "flexShrink": 1,
          "fontSize": 18,
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;
